src/Diagnostics.RuntimeHost/Services/CacheService/DiagEntityTableCacheService.cs

```diff
@@ -16,11 +17,34 @@ public class DiagEntityTableCacheService : IDiagEntityTableCacheService
 
         private IStorageService storageService;
 
+        int cacheExpirationTimeInSecs = 60;
+
         public DiagEntityTableCacheService(IStorageService service)
         {
             cache = new ConcurrentDictionary<string, List<DiagEntity>>();
             storageService = service;
+            StartPolling();
         }
+
+        private async void StartPolling()
+        {
+            do
+            {
+                await Task.Delay(cacheExpirationTimeInSecs * 1000);
```

Comment:
QQ Rekha - What is the expected behavior if there was an exception in the logic below. 
Should we log the exception and still continue to poll. (and maybe add an email alert to begin with to see if there are exceptions here in PROD)

Reply:
If there is an exception while retrieving data from storage, the exception is logged and null is returned. We check for null before adding to cache. So cache wont be updated and polling continues.

Reply:
One more point to think about : Today our detector changes (or new detectors) light up in 10 sec. Correct me if I am wrong here but it looks like now it will take 60 sec to see the new detector after publish. While I think 10 sec is too aggressive for polling, maybe starting with 30 sec might not be a bad idea.